lib/semantics/tools.h

```diff
@@ -85,6 +86,10 @@ const Symbol *FindExternallyVisibleObject(
 
 bool ExprHasTypeCategory(
     const evaluate::GenericExprWrapper &expr, const common::TypeCategory &type);
+bool ExprHasTypeKind(const evaluate::GenericExprWrapper &expr, int kind);
+bool ExprTypeKindIsDefault(
+    const evaluate::GenericExprWrapper &expr, const SemanticsContext &context);
+bool ExprIsScalar(const evaluate::GenericExprWrapper &expr);
```

Comment:
If there are other places in the code that check rank > 0, you ought to update them to use this routine if appropriate.

Reply:
Not needed anymore, removed.
Adding new (unrelated) functions to tools exceeds the purpose of this PR.